File: katas/7 kyu/odd-or-even/solution.py
import codewars_test as test
# -------------------------------- my solution ------------------------------- #
def odd_or_even(arr):
    return 'even' if sum(arr) % 2 == 0 else 'odd'
# ----------------------------------- test ----------------------------------- #
@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(odd_or_even([0, 1, 2]), "odd")
        test.assert_equals(odd_or_even([0, 1, 3]), "even")

File: katas/7 kyu/odd-or-even/solution.php
<!-- odd-or-even -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function odd_or_even(array $a): string {
  return array_sum($a) % 2 == 0 ? 'even' : 'odd';
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function odd_or_even(array $a): string {
  return array_sum($a) % 2 ? 'odd' : 'even';
}
/* ---------------------------------- TEST ---------------------------------- */
class OddOrEvenTest extends TestCase {
  public function testDescriptionExamples() {
    $this->assertEquals('even', odd_or_even([0]));
    $this->assertEquals('odd', odd_or_even([2, 5, 34, 6]));
    $this->assertEquals('even', odd_or_even([0, -1, -5]));
  }
}